core/src/main/clojure/xtdb/util.clj

```diff
@@ -117,6 +117,18 @@
     (.position bb 0)
     bb))
 
+(defn byte-buffer->uuid [^ByteBuffer bb]
+  (UUID. (.getLong bb 0) (.getLong bb 1)))
+
+(defn arrow-buf-pointer->bb [^ArrowBufPointer ptr]
```

Comment:
this might well go if you go for the iid-buf -> ABP suggestion above, but if not, IIRC you can call `(.nioBuffer (.getBuf ptr) (.getOffset ptr) (.getLength ptr))`